pkgs/applications/editors/neovim/utils.nix

```diff
@@ -229,12 +229,36 @@ let
         ln -s ${grammar}/parser $out/parser/${name}.so
       '');
 
+  /*
+    Fork of vimUtils.packDir that additionnally generates a propagated-build-inputs-file that
+    can be used by the lua hooks to generate a proper LUA_PATH
+
+    Generates a packpath folder as expected by vim
+       Example:
+       packDir ( {myVimPackage = { start = [ vimPlugins.vim-fugitive ]; opt = []; }; })
+       => "/nix/store/xxxxx-pack-dir"
+  */
+  packDir = packages:
+  let
+    rawPackDir = vimUtils.packDir packages;
+
+  in
+    rawPackDir.override ({
+    postBuild = ''
+      mkdir $out/nix-support
+      for i in $(find -L $out -name propagated-build-inputs ); do
+        cat "$i" >> $out/nix-support/propagated-build-inputs
```

Comment:
question: if multiple packages have the same dependency, would this result in duplicate entries? 
If so, could this be problematic?

Reply:
this would result in duplicate entries. I think lua will load the first plugin on the LUA_PATH. 
Until we have lua lockfiles, if you have several different lua dependencies, I guess you are actively screwing up since in nixpkgs we only keep one version of each lua plugin.